redhat-appstudio/service-provider-integration-operator
# Introduce the SecretStorage interface that generalizes the token storage to store arbitrary byte arrays.

### What does this PR do?
Introduce the SecretStorage interface that generalizes the token storage to store arbitrary byte arrays.

The existing implementations of TokenStorage interface were converted to simple DefaultTokenStorage instances set up with appropriate SecretStorage implementations.


### What issues does this PR fix or reference?
https://issues.redhat.com/browse/SVPI-371

### How to test this PR?
There should be no noticeable difference in functionality.

```
quay.io/redhat-appstudio/pull-request-builds:spi-controller-517
quay.io/redhat-appstudio/pull-request-builds:spi-oauth-517
```
